storage/innobase/ut/ut0crc32.cc

```diff
@@ -210,7 +214,43 @@ ut_crc32_64_low_hw(
 
 	return(static_cast<uint32_t>(crc_64bit));
 }
+#elif (_WIN64)
+/** Calculate CRC32 over 8-bit data using a hardware/CPU instruction.
+@param[in,out]	crc	crc32 checksum so far when this function is called,
+when the function ends it will contain the new checksum
+@param[in,out]	data	data to be checksummed, the pointer will be advanced
+with 1 byte
+@param[in,out]	len	remaining bytes, it will be decremented with 1 */
+inline
+void
+ut_crc32_8_hw(
+	uint32_t*	crc,
+	const byte**	data,
+	ulint*		len)
+{
+	*crc = _mm_crc32_u8(*crc, (*data)[0]);
+	(*data)++;
+	(*len)--;
+}
```

Comment:
To avoid code duplication I'd integrate it into original function, something like:
```
ut_crc32_8_hw(
        uint32_t*       crc,
        const byte**    data,
        ulint*          len)
{
#if _WIN64
        *crc = _mm_crc32_u8(*crc, (*data)[0]);
#else
        asm("crc32b %1, %0"
            /* output operands */
            : "+r" (*crc)
            /* input operands */
            : "rm" ((*data)[0]));
#endif

        (*data)++;
        (*len)--;
}
```

Reply:
I actually like the original more than duplication of ifdefs

Reply:
There's hw crc32 implementation for ARM in the queue. I was going to reshuffle this code a bit. It is going to be simpler if we have code in more compact form.